src/python/ensembl/database/dbconnection.py

```diff
@@ -0,0 +1,202 @@
+    # You can access the database data via sql queries, for instance:
+    results = dbc.execute('SELECT * FROM my_table;')
+    # Or via a connection in a transaction manner:
+    with dbc.begin() as conn:
+        results = conn.execute('SELECT * FROM my_table;')
+
+"""
+
+import contextlib
+from typing import Dict, List, TypeVar
+
+import sqlalchemy
+from sqlalchemy import create_engine, event, text
+from sqlalchemy.orm.session import Session
+from sqlalchemy.sql import select
+
+
+# Create the Query type as an alias for all the possible types an SQL query can be stored into
+Query = TypeVar('Query', str, sqlalchemy.sql.expression.ClauseElement)
+# Create the URL type as an alias for all the possible types an URL can be stored into
+URL = TypeVar('URL', str, sqlalchemy.engine.url.URL)
+
+
+class DBConnection:
+    """Database connection handler, providing also the database's schema and properties.
+
+    Args:
+        url: URL to the database, e.g. ``mysql://user:passwd@host:port/my_db``.
+
+    """
+    def __init__(self, url: URL) -> None:
+        self._engine = create_engine(url)
+        self.load_metadata()
+
+    def __repr__(self) -> str:
+        """Returns a string representation of this object."""
+        return f'{self.__class__.__name__}({self.url!r})'
+
+    def load_metadata(self) -> None:
+        """Loads the metadata information of the database."""
+        # Note: Just reflect() is not enough as it would not delete tables that no longer exist
+        self._metadata = sqlalchemy.MetaData(bind=self._engine)
+        self._metadata.reflect()
+
+    @property
+    def url(self) -> str:
+        """Database URL."""
+        return str(self._engine.url)
+
+    @property
+    def db_name(self) -> str:
+        """Database name."""
+        return self._engine.url.database
+
+    @property
+    def host(self) -> str:
+        """Database host."""
+        return self._engine.url.host
+
+    @property
+    def dialect(self) -> str:
+        """SQLAlchemy database dialect of the database host."""
+        return self._engine.name
+
+    @property
+    def tables(self) -> Dict[str, sqlalchemy.schema.Table]:
+        """Dictionary of :class:`~sqlalchemy.schema.Table` objects keyed to their name."""
+        return self._metadata.tables
+
+    def get_primary_key_columns(self, table: str) -> List[str]:
+        """Returns the list of primary key column names for the given table.
+
+        Args:
+            table: Table name.
+
+        """
+        return [col.name for col in self.tables[table].primary_key.columns.values()]
+
+    def get_columns(self, table: str) -> List[str]:
+        """Returns the list of column names for the given table.
+
+        Args:
+            table: Table name.
+
+        """
+        return [col.name for col in self.tables[table].columns]
+
+    @property
+    def schema_type(self) -> str:
+        """Schema type of the database, located in the ``meta`` table."""
```

Comment:
What exception is raised if the table `meta` doesn't exist or if the row is not found ? (I guess `IndexError` for the latter). I think it would be worth documenting